### DISCUSSION TITLE
Thermal Model and Charging Simulation

### DISCUSSION THREAD

**Discussion by glogin0424** [pybamm-team/PyBaMM#3491] [Q&A]
Hi everyone,

I'm currently working on optimal initial battery temperature to start fast charging in cold weather. I need some help with thermal modeling simulation.

I'm trying to simulate different thermal environment while sweeping through different initial temperature to check the 'negative electrode surface potential vs surface to prevent li-plating'. By thermal environment, I mean setting different ambient temperature or changing heat transfer coefficient. Initial temperature here refers to the ambient temperature after the discharge experiment and the cell temperature when the charging experiment starts. I reset the ambient temperature for charging experiment. Through this simulation, my goal is to determine the maximum c-rate where anode potential is always above 0V vs Li/Li+ at different initial temperature and thermal environment. 

I am using 'OKane2022' battery parameters with DFN model. The battery is fully discharged and then simulated with different c-rates. C-rates vary from C/4 to 4C.

I initially tried simulating 'thermal model' option without adding any thermal parameters and realized that the cell temperature goes up to 80˚C at 4C. To cool down the battery cell, I added cooling plate parameters to the thermal model by setting the coolant surface area and also increasing the heat transfer coefficient to 5. However, I'm seeing that regardless of different initial temperature, my maximum current (c-rate where minimum anode potential is above 0) does not change. Below figure shows how I obtain the maximum c-rate. I swept through different initial temperature but the results are all the same.

<img width="408" alt="figure" src="https://github.com/pybamm-team/PyBaMM/assets/32934086/6fa231ed-d8e1-4bec-a47e-8adb6e37e7d5">

My first question is, is the way I'm setting the initial temperature and ambient temperature correct for the charging experiment? My second question is, am I setting the thermal parameters correctly? It does not make sense that different initial temperature charging does not affect maximum c-rate. I think if you start at higher temperature, you should be able to charge at higher c-rate but the simulation result shows not. 

Below is the code I used for my simulation. I excluded the code block I used to get maximum c-rate due to code size but I can include it as well.  

```python3
import pybamm 

#set model
options={"lithium plating":"partially reversible", "thermal": "lumped"}
model = pybamm.lithium_ion.DFN(options)

# set model paramters
parameter_values = pybamm.ParameterValues("OKane2022")
parameter_values.update({"Upper voltage cut-off [V]": 4.21})
parameter_values.update({"Lithium plating transfer coefficient": 0.5})
parameter_values.update({"Dead lithium decay constant [s-1]": 1E-4})
A = parameter_values["Electrode width [m]"] * parameter_values["Electrode height [m]"]
parameter_values.update(
    {
        "Total heat transfer coefficient [W.m-2.K-1]": 5,
        "Cell cooling surface area [m2]": 2 * A,
    }
)
init_temp = 308.15
parameter_values.update({"Ambient temperature [K]": init_temp})

# discharge experiment
s = pybamm.step.string
experiment_discharge = pybamm.Experiment(
    [
        (s("Discharge at C/20 until 2.5 V", period="10 minutes"),
        s("Rest for 1 hour", period="3 minutes")),
    ]
)
sim_discharge = pybamm.Simulation(model, parameter_values=parameter_values, experiment=experiment_discharge)
sol_discharge = sim_discharge.solve(calc_esoh=False)
model.set_initial_conditions_from(sol_discharge, inplace=True)

# charging experiment
C_rates = ["C/8", "C/4", "C/2", "1C", "1.5C", "2C", "2.5C", "3C", "3.5C", "4C"]
experiments = {}
for C_rate in C_rates:
    experiments[C_rate] = pybamm.Experiment(
    [
        (f"Charge at {C_rate} until 4.2 V",
        "Hold at 4.2 V until C/20",
        "Rest for 1 hour")
    ]
)
def define_and_solve_sims(model, experiments, parameter_values):
    sims = {}
    for C_rate, experiment in experiments.items():
        sim = pybamm.Simulation(model, experiment=experiment, parameter_values=parameter_values)
        # sim.solve(calc_esoh=False)
        sim.solve(calc_esoh=False, initial_soc=0.5)
        sims[C_rate] = sim

    return sims
amb_temp = 308.15
parameter_values.update({"Ambient temperature [K]": amb_temp}) #set to different ambient temp
sims_partial = define_and_solve_sims(model, experiments, parameter_values)
sols_partial = {}
for C_rate in C_rates:
  sols_partial[C_rate] = sims_partial[C_rate].solution

```

I am new to battery simulation and modeling so some help from the community would be appreciated!

Thank you so much.

**Comment by brosaplanella** [comment]
It's strange as the parameters depend on temperature so an effect should be observed. I tried running your code but did not produce the plot you shared, can you share a minimum working example that produces that plot, so we can test at our end if temperature changes?

Also, just to double check, which PyBaMM version are you using?
